### PR TITLE
Upgrade from 0.4.0 to 0.5.0

### PR DESCRIPTION
**Pull request by jcrossley3**

### CODE REVIEW

File: deploy/resources/knative-build-0.5.0.yaml [hunk 1]
```diff
@@ -431,8 +440,11 @@ spec:
           valueFrom:
             fieldRef:
               fieldPath: metadata.namespace
-        image: quay.io/openshift-knative/knative-build-webhook:v0.4.0
```
Comment (matzew) [inline deploy/resources/knative-build-0.5.0.yaml]: shouldn't we here continue to use our `quay.io` images ?

File: README.md [hunk 2]
```diff
@@ -5,17 +5,58 @@ If you don't already have it, install
...
+
+    ./hack/extract-crds.sh
+
+Update [deploy/role.yaml](deploy/role.yaml/) to match the RBAC
+policies in the upstream manifest[s].
+
+And then generate a basic `ClusterServiceVersion` passing in a version
+that corresponds to those manifests:
+
+    operator-sdk olm-catalog gen-csv --csv-version 0.5.1
```
Comment (matzew) [inline README.md]: the build is at `0.5.0`  - I think it's better to use `$VERSION` in here ?

File: deploy/role.yaml [hunk 3]
```diff
@@ -52,3 +52,85 @@ rules:
...
+  - apiGroups: ["caching.internal.knative.dev"]
+    resources: ["images"]
+    verbs: ["get", "list", "create", "update", "delete", "deletecollection", "patch", "watch"]
+  - apiGroups: ["policy"]
+    resources: ["podsecuritypolicies"]
+    resourceNames: ["knative-build"]
+    verbs: ["use"]
+
+  # The above rules are from upstream. The remaining are
+  # required for either OpenShift or our operator
```
Comment (matzew) [inline deploy/role.yaml]: these :arrow_down: are generic to the different operators, than, right ?

File: deploy/role.yaml [hunk 4]
```diff
@@ -52,3 +52,85 @@ rules:
...
+    resources: ["builds/status", "buildtemplates/status", "clusterbuildtemplates/status"]
+    verbs: ["get", "list", "create", "update", "delete", "patch", "watch"]
+  - apiGroups: ["caching.internal.knative.dev"]
+    resources: ["images"]
+    verbs: ["get", "list", "create", "update", "delete", "deletecollection", "patch", "watch"]
+  - apiGroups: ["policy"]
+    resources: ["podsecuritypolicies"]
+    resourceNames: ["knative-build"]
+    verbs: ["use"]
+
```
Comment (matzew) [inline deploy/role.yaml]: :arrow_up:  are basically those from the `200-clusterrole.yaml` file, upstream

File: deploy/olm-catalog/knative-build-operator/0.4.0/knative-build-operator.v0.4.0.clusterserviceversion.yaml [hunk 5]
```diff
@@ -0,0 +1,293 @@
...
+                  valueFrom:
+                    fieldRef:
+                      fieldPath: metadata.namespace
+                - name: POD_NAME
+                  valueFrom:
+                    fieldRef:
+                      fieldPath: metadata.name
+                - name: OPERATOR_NAME
+                  value: knative-build-operator
+                image: quay.io/jcrossley3/knative-build-operator:v0.4.0
```
Comment (matzew) [inline deploy/olm-catalog/knative-build-operator/0.4.0/knative-build-operator.v0.4.0.clusterserviceversion.yaml]: @jcrossley3  org ?

File: deploy/olm-catalog/knative-build-operator/0.5.0/knative-build-operator.v0.5.0.clusterserviceversion.yaml [hunk 6]
```diff
@@ -0,0 +1,297 @@
...
+                  valueFrom:
+                    fieldRef:
+                      fieldPath: metadata.name
+                - name: OPERATOR_NAME
+                  value: knative-build-operator
+                image: quay.io/openshift-knative/knative-build-operator:v0.5.0
+                args:
+                  - --olm
+                  - --install
+                  - --filename=https://raw.githubusercontent.com/openshift/knative-build/release-v0.5.0/openshift/release/knative-build-v0.5.0.yaml
```
Comment (matzew) [inline deploy/olm-catalog/knative-build-operator/0.5.0/knative-build-operator.v0.5.0.clusterserviceversion.yaml]: the readme talks about `args: ["--olm", "--install"]`; should it add the `--filename` ?

File: deploy/resources/knative-build-0.5.0.yaml (hunk 1 above)
Comment (jcrossley3) [reply deploy/resources/knative-build-0.5.0.yaml]: No. I'm thinking we always embed the official upstream release in the image at build-time (deploy/resources) and we use the `--filename` arg to override at runtime.

File: README.md (hunk 2 above)
Comment (jcrossley3) [reply README.md]: good call

File: deploy/role.yaml (hunk 3 above)
Comment (jcrossley3) [reply deploy/role.yaml]: I think they may be specific to each operator, but i'm not sure.

File: deploy/role.yaml (hunk 4 above)
Comment (jcrossley3) [reply deploy/role.yaml]: yes, and feel free to move them if it makes more sense.

File: deploy/olm-catalog/knative-build-operator/0.5.0/knative-build-operator.v0.5.0.clusterserviceversion.yaml (hunk 6 above)
Comment (jcrossley3) [reply deploy/olm-catalog/knative-build-operator/0.5.0/knative-build-operator.v0.5.0.clusterserviceversion.yaml]: Maybe? It should at least mention it, probably.

**Comment by jcrossley3** [comment]
Oops! Dammit, I didn't mean to do that yet. :)

**Pull request merged**